src/main/java/cubicchunks/server/CubeIO.java

```diff
@@ -363,7 +362,7 @@ private Cube readCubeFromNbtAndAddToColumn(Column column, final int cubeX, final
 		final Cube cube = column.getOrCreateCube(cubeY, false);
 
 		// get the worldgen stage
-		cube.setGeneratorStage(GeneratorStage.values()[nbt.getByte("GeneratorStage")]);
+		cube.setGeneratorStage(world.getGeneratorPipeline().getStage(nbt.getByte("GeneratorStage")));
```

Comment:
If the GeneratorPipeline is guaranteed to not be altered, using ordinals is fine. Using strings might allow for easier compatibility later on.